Ui/Component/Listing/Column/MenuList/PageActions.php

```diff
@@ -1,26 +1,93 @@
 <?php
+
+declare(strict_types=1);
+
 namespace Snowdog\Menu\Ui\Component\Listing\Column\MenuList;
 
 use Magento\Ui\Component\Listing\Columns\Column;
+use Magento\Framework\View\Element\UiComponent\ContextInterface;
+use Magento\Framework\View\Element\UiComponentFactory;
+use Magento\Framework\UrlInterface;
 
+/**
+ * Class PageActions
+ *
+ * Adds action buttons into actions column
+ */
 class PageActions extends Column
 {
-    public function prepareDataSource(array $dataSource)
+    /**
+     * Url path
+     */
+    const URL_PATH_EDIT = 'snowmenu/menu/edit';
+    const URL_PATH_DELETE = 'snowmenu/menu/delete';
+
+    /**
+     * @var UrlInterface
+     */
+    private $urlBuilder;
+
+    /**
+     * @param ContextInterface $context
+     * @param UiComponentFactory $uiComponentFactory
+     * @param UrlInterface $urlBuilder
+     * @param array $components
+     * @param array $data
+     */
+    public function __construct(
+        ContextInterface $context,
+        UiComponentFactory $uiComponentFactory,
+        UrlInterface $urlBuilder,
+        array $components = [],
+        array $data = []
+    ) {
+        $this->urlBuilder = $urlBuilder;
+        parent::__construct(
+            $context,
+            $uiComponentFactory,
+            $components,
+            $data
+        );
+    }
+
+    /**
+     * @param array $dataSource
+     * @return array
+     */
+    public function prepareDataSource(array $dataSource): array
     {
-        if (isset($dataSource["data"]["items"])) {
-            foreach ($dataSource["data"]["items"] as & $item) {
-                $name = $this->getData("name");
-                $id = "X";
-                if (isset($item["menu_id"])) {
-                    $id = $item["menu_id"];
+        if (isset($dataSource['data']['items'])) {
+            foreach ($dataSource['data']['items'] as & $item) {
+                if (isset($item['menu_id'])) {
+                    $name = $this->getData('name');
+                    $item[$name] = [
+                        'edit' => [
```

Comment:
This method seems pretty long, what do you think if we extract `edit` and `delete` buttons generation into 2 different private methods so this part will look like:

```php
// ...
$item[$name] = [
    'edit' => $this->getEditButton($item['menu_id']),
    'delete' => $this->getDeleteButton($item['menu_id']),    
]
// ...
```

Reply:
Done